ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/Uniform.hs

```diff
@@ -70,7 +69,13 @@ tests =
     adjustQuickCheckTests (`div` 10) $
     testProperty "serve adversarial branches" prop_serveAdversarialBranches,
     adjustQuickCheckTests (`div` 100) $
-    testProperty "the LoE stalls the chain, but the immutable tip is honest" prop_loeStalling
+    testProperty "the LoE stalls the chain, but the immutable tip is honest" prop_loeStalling,
+    adjustQuickCheckTests (`div` 100) $
+    -- This is a crude way of ensuring that we don't get chains with more than 100 blocks,
+    -- because this test writes the immutable chain to disk and `instance Binary TestBlock`
+    -- chokes on long chains.
```

Comment:
It looks to me like we could improve the `instance Binary TestBlock`. At the very least, in our use-case, a run-length encoding would push the limit quite a lot. There are many other clever encodings we could imagine to push the limit. Maybe not the scope of this PR, though.

Reply:
could be done with reflection (`Given` constraint) but would be too infectious to handle in this PR